rest_framework/fields.py

```diff
@@ -54,12 +54,17 @@ def is_simple_callable(obj):
         """
         True if the object is a callable that takes no arguments.
         """
-        if not callable(obj):
+        if not (inspect.isfunction(obj) or inspect.ismethod(obj)):
             return False
 
         sig = inspect.signature(obj)
         params = sig.parameters.values()
-        return all(param.default != param.empty for param in params)
+        return all(
+            param.kind == param.VAR_POSITIONAL or
+            param.kind == param.VAR_KEYWORD or
+            param.default != param.empty
+            for param in params
+        )
```

Comment:
@carltongibson - it probably wouldn't have hurt to have added a second set of parens wrapping the conditions

```py
return all((
        param.kind == param.VAR_POSITIONAL or
        param.kind == param.VAR_KEYWORD or
        param.default != param.empty
    ) for param in params
)
```